profile.d/00-pengwin.sh

```diff
@@ -21,6 +21,7 @@ if [ -n "${WSL_INTEROP}" ]; then
 
     wsl2_d_tmp="$(eval "$ipconfig_exec" | sed "$((wsl2_d_tmp - 4))"','"$((wsl2_d_tmp + 0))"'!d' | grep IPv4 | cut -d : -f 2 | sed -e "s|\s||g" -e "s|\r||g")"
     export DISPLAY=${wsl2_d_tmp}:0
+    export PULSE_SERVER=${wsl2_d_tmp}
```

Comment:
tcp: shouldn't be inserted before the ip?

Reply:
just the ip address is enough. 

I disabled a lot of stuff in the default pa package but it still depends on avahi, jack, gsettings not sure if you want to pursue stripping libpulse to that effect.  

to daemonize the PA 1.1 server for windows you need to edit daemon.conf `exit-idle-time = 20` to ` exit-idle-time = -1`
don't get me started on the WSL2 networking q.q default.pa `load-module module-native-protocol-tcp auth-anonymous=1` could explore more deeply into auth.  I DO have to enable the pulseaudio firewall exception for TCP

Reply:
perhaps there is a way to itterate the firewall to match the WSL IP facing windows??

Reply:
`pulseaudio		Private	Yes	Allow	No	C:\wsl\bin\pulseaudio.exe	Any	Local subnet	TCP	4713	Any	Any	Any	Any	Any	Any`

Reply:
> this allows any machine on the network to send audio to you. If you are not firewalled from the internet or you don’t consider your network secure, you must use a cookie file that PulseAudio generates for you. You can find it in your user profile: on Windows XP, it’s located in C:\Documents and Settings\<your username>\.pulse-cookie. Vista and 7 users can find it at C:\Users\<your username>\.pulse-cookie. Simply copy this file to the home directory of the user that is sending you sound on the remote machine. Often this will be your own home directory, but some daemons run their own (for example, MPD on Ubuntu requires the cookie to be in /var/lib/mpd). When you’ve done this, you can remove the “auth-anonymous=1” part from the line above.

https://parseq.co.uk/wordpress/archives/setting-up-pulseaudio-1-0-beta-for-windows

bam auth done firewall done

Reply:
bat:
```
@echo off
cd C:\wsl
C:\wsl\bin\pulseaudio.exe
```
vbs:
```
Option Explicit
Dim PathApplication,ShortcutName,VbsPath
VbsPath = Wscript.ScriptFullName
PathApplication = "C:\wsl\share\windows\pulseaudio.bat"
ShortcutName = "pulseaudio"
Call Shortcut(VbsPath,ShortcutName)
Call Hidden_Run(Dblquote(PathApplication))
'*********************************************************************************
Sub Shortcut(PathApplication,ShortcutName)
    Dim objShell,StartFolder,objShortCut,MyTab
    Set objShell = CreateObject("WScript.Shell")
    MyTab = Split(PathApplication,"\")
    If ShortcutName = "" Then
        ShortcutName = MyTab(UBound(MyTab))
    End if
    StartFolder = objShell.SpecialFolders("Startup")
    Set objShortCut = objShell.CreateShortcut(StartFolder & "\" & ShortcutName & ".lnk")
    objShortCut.TargetPath = Dblquote(PathApplication)
    ObjShortCut.IconLocation = "%SystemRoot%\system32\SHELL32.dll,-25"
    objShortCut.Save
End Sub
'*********************************************************************************
Function DblQuote(Str)
    DblQuote = Chr(34) & Str & Chr(34)
End Function
'*********************************************************************************
Function Hidden_Run(MyProgram)
    Dim ws,Result
    Set ws = CreateObject("wscript.Shell")
    Result = ws.run(MyProgram,0,True) '0 to hide the program
    Hidden_Run = Result
End Function
'*********************************************************************************
```